File: easy/70_ClimbingStairs.py
# 70. Climbing Stairs
# Easy
# https://leetcode.com/problems/climbing-stairs/

class Solution:
    def climbStairs(self, n: int) -> int:
        dp = []
        dp.append(1)
        dp.append(2)
        for i in range(n - 2):
            dp.append(dp[i] + dp[i + 1])
        return dp[n - 1]

File: easy/70_ClimbingStairs.rs
// 70. Climbing Stairs
// easy
// https://leetcode.com/problems/climbing-stairs

impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        if n == 1 {
            return 1;
        }

        let mut dp_0 = 1;
        let mut dp_1 = 2;
        for i in (2..n) {
            let dp = dp_0 + dp_1;
            dp_0 = dp_1;
            dp_1 = dp;
        }

        dp_1
    }
}